app/models/route.rb

```diff
@@ -1,6 +1,20 @@
 class Route < ActiveRecord::Base
   validates :name, presence: true
-  has_many :railway_station_routes
-  has_many :railway_stations, :through => :railway_station_routes
+  validate :station_count
+
+  has_many :railway_stations_routes
+  has_many :railway_stations, through: :railway_stations_routes
   has_many :trains
+
+  before_validation :set_name
+
+  private
+
+  def set_name
+    self.name = "#{railway_stations.first.title}-#{railway_stations.last.title}"
+  end
```

Comment:
you need to catch the error here if there are no stations at all. Add check rubocop please.